src/prefect/engine.py

```diff
@@ -2117,6 +2114,18 @@ async def tick():
     # flag to ensure we only update the task run name once
     run_name_set = False
 
+    run_on_completion_hooks_on_cached = (
+        PREFECT_RUN_ON_COMPLETION_HOOKS_ON_CACHED and state.is_cached()
+    )
+
+    if run_on_completion_hooks_on_cached:
+        task_run = await client.read_task_run(task_run.id)
```

Comment:
hmm, do we definitely need to make an API call to get the `task_run` here? it seems in scope based on 2107

Reply:
Good catch!